### PR TITLE
Update tigris starter ts

### PR DESCRIPTION
**Pull request by icode247**

### CODE REVIEW

File: src/db/product.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {
+  Field,
+  PrimaryKey,
+  TigrisCollection,
+  TigrisDataTypes,
+} from "@tigrisdata/core";
+
+@TigrisCollection("products02")
+export class Product {
+  @PrimaryKey(TigrisDataTypes.STRING, { order: 1, autoGenerate: true })
```
Comment (ovaistariq) [inline src/db/product.ts:10]: Change this to int64 just like you have in user model

File: src/db/order.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  @PrimaryKey(TigrisDataTypes.STRING, { order: 1 })
+  productId: string;
+
+  @Field(TigrisDataTypes.INT32)
+  quantity: number;
+}
+
+@TigrisCollection("orders")
+export class Order {
+  @PrimaryKey(TigrisDataTypes.STRING, { order: 1 })
```
Comment (ovaistariq) [inline src/db/order.ts:18]: Change this to int64 just like you have in user model

File: src/db/product.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {
+  Field,
+  PrimaryKey,
+  TigrisCollection,
+  TigrisDataTypes,
+} from "@tigrisdata/core";
+
+@TigrisCollection("products02")
```
Comment (ovaistariq) [inline src/db/product.ts:8]: products

File: src/db/order.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import {
+  Field,
+  PrimaryKey,
+  TigrisCollection,
+  TigrisDataTypes,
+} from "@tigrisdata/core";
+
+export class Product {
```
Comment (ovaistariq) [inline src/db/order.ts:8]: You don't need this duplicate class. Use the in in product model

**Review by ovaistariq — CHANGES_REQUESTED** [review]

File: src/db/order.ts [hunk 5]
```diff
@@ -4,19 +4,12 @@ import {
...
-  quantity: number;
-}
+import { Product } from "./product";
 
 @TigrisCollection("orders")
 export class Order {
-  @PrimaryKey(TigrisDataTypes.STRING, { order: 1 })
-  orderId: string;
+  @PrimaryKey(TigrisDataTypes.INT64, { order: 1 })
+  orderId: number;
```
Comment (ovaistariq) [inline src/db/order.ts:12]: Use `string` with `int64`

File: src/db/product.ts [hunk 6]
```diff
@@ -5,10 +5,10 @@ import {
   TigrisDataTypes,
 } from "@tigrisdata/core";
 
-@TigrisCollection("products02")
+@TigrisCollection("products")
 export class Product {
-  @PrimaryKey(TigrisDataTypes.STRING, { order: 1, autoGenerate: true })
-  productId: string;
+  @PrimaryKey(TigrisDataTypes.INT64, { order: 1, autoGenerate: true })
+  productId: number;
```
Comment (ovaistariq) [inline src/db/product.ts:11]: Use `string` with `int64`

File: src/controllers/user-controller.ts [hunk 7]
```diff
@@ -19,7 +18,9 @@ export class UserController implements Controller {
   public getUser = async (req: Request, res: Response, next: NextFunction) => {
     this.users
       .findOne({
-        userId: Number.parseInt(req.params.id),
+        filter: {
+          userId: Number.parseInt(req.params.id),
```
Comment (ovaistariq) [inline src/controllers/user-controller.ts:22]: No need to parse as int, just like in the product and order classes above

File: src/db/user.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  Field,
+  PrimaryKey,
+  TigrisCollection,
+  TigrisDataTypes,
+} from "@tigrisdata/core";
+
+@TigrisCollection("users")
+export class User {
+  @PrimaryKey(TigrisDataTypes.INT64, { order: 1, autoGenerate: true })
+  userId: number;
```
Comment (ovaistariq) [inline src/db/user.ts:11]: use `string` with int64

File: src/db/models/product.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  Field,
+  PrimaryKey,
+  TigrisCollection,
+  TigrisDataTypes,
+} from "@tigrisdata/core";
+
+@TigrisCollection("products")
+export class Product {
+  @PrimaryKey(TigrisDataTypes.INT64, { order: 1, autoGenerate: true })
+  productId: string;
```
Comment (ovaistariq) [inline src/db/models/product.ts:11]: Mark this as optional as well as the value will be auto-generated.

File: src/db/models/user.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  Field,
+  PrimaryKey,
+  TigrisCollection,
+  TigrisDataTypes,
+} from "@tigrisdata/core";
+
+@TigrisCollection("users")
+export class User {
+  @PrimaryKey(TigrisDataTypes.INT64, { order: 1, autoGenerate: true })
+  userId: string;
```
Comment (ovaistariq) [inline src/db/models/user.ts:11]: Mark this as optional as well as the value will be auto-generated.

File: src/db/models/order.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+} from "@tigrisdata/core";
+import { Product } from "./product";
+
+@TigrisCollection("orders")
+export class Order {
+  @PrimaryKey(TigrisDataTypes.INT64, { order: 1, autoGenerate: true })
+  orderId?: string;
+
+  @Field(TigrisDataTypes.INT64)
+  userId: number;
```
Comment (ovaistariq) [inline src/db/models/order.ts:15]: This will also be a string because it refers to the userId field in the user model.

Comment (icode247) [reply src/db/models/order.ts]: okay, fixed

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**